src/spox/_attributes.py

```diff
@@ -184,38 +200,56 @@ def maybe(
     ) -> Optional[AttrIterableT]:
         return cls(tuple(value)) if value is not None else None
 
+    def _to_onnx_deref(self, key: str) -> AttributeProto:
+        return make_attribute(key, self.value, attr_type=self._attribute_proto_type_int)
+
 
 class AttrFloat32s(_AttrIterable[float]):
     _attribute_proto_type_int = AttributeProto.FLOATS
 
     def _to_onnx_deref(self, key: str) -> AttributeProto:
-        # ensure values are all floats
-        values = [float(v) for v in self.value]
-        return make_attribute(key, values, attr_type=AttributeProto.FLOATS)
+        try:
+            transformed = tuple(float(v) for v in self.value)
+        except ValueError as e:
+            raise AttributeTypeError("Attribute values don't seem to be floats.") from e
+        return make_attribute(
+            key, transformed, attr_type=self._attribute_proto_type_int
+        )
 
 
 class AttrInt64s(_AttrIterable[int]):
     _attribute_proto_type_int = AttributeProto.INTS
 
-    def _to_onnx_deref(self, key: str) -> AttributeProto:
-        return make_attribute(key, self.value, attr_type=AttributeProto.INTS)
-
 
 class AttrStrings(_AttrIterable[str]):
     _attribute_proto_type_int = AttributeProto.STRINGS
 
     def _to_onnx_deref(self, key: str) -> AttributeProto:
+        try:
+            transformed = tuple(v.encode() for v in self.value)
```

Comment:
Same as above. If we provide a `str` then `make_attribute` will call `encode` for us. If something goes wrong then we get an error from onnx/protobuf with a pretty useless error message. At that point, we can either phrase our error message as a suggestion, or we only iterate through all elements if there was indeed an error (which should be very rare) to provide the user with a more definitive error message.

Reply:
This will make me happier too :) Check https://github.com/Quantco/spox/pull/122/commits/09df28f1faa2539e1bf60cd2ddf595b092192925